File: laba 4 py/4.3/6.py
from Lenght import lenght
text = input("Введите предложение:  ")
count = 0
for x in range(lenght(text)-1):
    x1 = text[x]
    x2 = text[x+1]
    if x1 == x2:
        count += 1
print(count)

File: laba 1 sem/laba 4/laba 4 pas/4.3/6.pas
function lenght(str: string): integer;
  var 
    i: integer;
    s: string;
  begin    
    i := 1;
    while s <> str do
      begin
      s += str[i];
      i += 1
    end;
  Result := i - 1;
end;
var
  text: string;
  count: Integer;
  x: Integer;
  x1, x2: char;

begin
  Write('Введите предложение: ');
  ReadLn(text);

  count := 0;
  for x := 1 to lenght(text) - 1 do
  begin
    x1 := text[x];
    x2 := text[x + 1];
    if x1 = x2 then
      count := count + 1;
  end;
  WriteLn(count);
end.